api/src/main/java/org/openmrs/module/clientregistry/api/search/CRSearchBundleProvider.java

```diff
@@ -2,26 +2,15 @@
 
 import java.io.Serializable;
 import java.util.List;
+
 import org.hl7.fhir.instance.model.api.IBaseResource;
-import org.openmrs.module.fhir2.FhirConstants;
-import org.openmrs.module.fhir2.api.FhirGlobalPropertyService;
+
 import ca.uhn.fhir.rest.server.SimpleBundleProvider;
 
 public class CRSearchBundleProvider extends SimpleBundleProvider implements Serializable {
-	
-	private final FhirGlobalPropertyService globalPropertyService;
-	
-	public CRSearchBundleProvider(List<? extends IBaseResource> patientList, FhirGlobalPropertyService globalPropertyService) {
+
+	public CRSearchBundleProvider(List<? extends IBaseResource> patientList) {
 		super(patientList);
-		this.globalPropertyService = globalPropertyService;
 	}
-	
-	@Override
-	public Integer preferredPageSize() {
-		if (size() == null) {
-			setSize(globalPropertyService.getGlobalProperty(FhirConstants.OPENMRS_FHIR_DEFAULT_PAGE_SIZE, 10));
-		}
-		return size();
-	}
```

Comment:
So from what you gathered, you mean this is never invoked?

Reply:
@samuelmale  This is invoked but never has an impact since the size will always be set via the superclass. So I am fixing this by adding the count param earlier before the query is made. See [here](https://github.com/openmrs/openmrs-module-clientregistry/pull/22/files/f26a0bdf6e0cdd1c0d7ca9e816fedb028314db41#diff-8f52abbb31939939d80d661c8ea67e3dae5fb15101ead920126a259d828b3e1fR101)